src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -35,7 +35,55 @@ class AladdinUserFaultType(Enum):
     InvalidAccountName = 'InvalidAccountName'
 
 
-class CommandRecommender():
+def get_error_type(error_msg):
+    """The the error type of the failed command from the error message.
+    The error types are only consumed by aladdin service for better recommendations.
+    """
+
+    error_type = AladdinUserFaultType.Unknown
+    if not error_msg:
+        return error_type.value
+
+    error_msg = error_msg.lower()
+    if 'unrecognized' in error_msg:
+        error_type = AladdinUserFaultType.UnrecognizedArguments
+    elif 'expected one argument' in error_msg or 'expected at least one argument' in error_msg \
+            or 'value required' in error_msg:
+        error_type = AladdinUserFaultType.ExpectedArgument
+    elif 'misspelled' in error_msg:
+        error_type = AladdinUserFaultType.UnknownSubcommand
+    elif 'arguments are required' in error_msg or 'argument required' in error_msg:
+        error_type = AladdinUserFaultType.MissingRequiredParameters
+        if '_subcommand' in error_msg:
+            error_type = AladdinUserFaultType.MissingRequiredSubcommand
+        elif '_command_package' in error_msg:
+            error_type = AladdinUserFaultType.UnableToParseCommandInput
+    elif 'not found' in error_msg or 'could not be found' in error_msg \
+            or 'resource not found' in error_msg:
+        error_type = AladdinUserFaultType.AzureResourceNotFound
+        if 'storage_account' in error_msg or 'storage account' in error_msg:
+            error_type = AladdinUserFaultType.StorageAccountNotFound
+        elif 'resource_group' in error_msg or 'resource group' in error_msg:
+            error_type = AladdinUserFaultType.ResourceGroupNotFound
+    elif 'pattern' in error_msg or 'is not a valid value' in error_msg or 'invalid' in error_msg:
```

Comment:
Is it enough to just match `pattern` instead of somethign like `invalid pattern`

Reply:
This are the rules provided by Aladdin Team for their internal error categories used to train model. I think it's ok as long as it make sense to them.